packages/cubejs-schema-compiler/src/adapter/BaseQuery.js

```diff
@@ -1976,14 +2040,37 @@ export class BaseQuery {
   }
 
   checkShouldBuildJoinForMeasureSelect(measures, keyCubeName) {
+    // When member expression references view, it would have to collect join hints from view
+    // Consider join A->B, as many-to-one, so B is multiplied and A is not, and member expression like SUM(AB_view.dimB)
+    // Both `collectCubeNamesFor` and `collectJoinHintsFor` would return too many cubes here
+    // They both walk join hints, and gather every cube present there
+    // For view we would get both A and B, because join hints would go from join tree root
+    // Even though expression references only B, and should be OK to use it with B as keyCube
+    // So this check would build new join tree from both A and B, B will be multiplied, and that would break check
+
     return measures.map(measure => {
-      const cubes = this.collectFrom([measure], this.collectCubeNamesFor.bind(this), 'collectCubeNamesFor');
-      const joinHints = this.collectFrom([measure], this.collectJoinHintsFor.bind(this), 'collectJoinHintsFor');
+      const memberNamesForMeasure = this.collectFrom(
+        [measure],
+        this.collectMemberNamesFor.bind(this),
+        'collectMemberNamesFor',
+      );
+
+      const nonViewMembers = memberNamesForMeasure
+        .filter(mem => {
+          const cubeName = this.cubeEvaluator.parsePathAnyType(mem)[0];
+          const cubeDef = this.cubeEvaluator.getCubeDefinition(cubeName);
+          return !cubeDef.isView;
+        })
+        .map(m => this.memberInstanceByPath(m));
```

Comment:
Maybe it would be better to use `ownedByCube` here?
Something like
```
const nonViewMembers = memberNamesForMeasure
         .map(m => this.memberInstanceByPath(m))
        .filter(mem => {
          
          return mem.ownedByCube;
        });     
```